File: crates/codemap-index/tests/fixtures/sample.ts
import express from "express";
import { normalize } from "./util";

const app = express();

export class ChatService {
  streamResponse(req: Request) {
    const data = normalize(req);
    return this.finish(data);
  }
  finish(data: unknown) { return data; }
}

app.post("/chat/stream", function chatStream(req, res) {
  const svc = new ChatService();
  return svc.streamResponse(req);
});

File: crates/codemap-index/tests/fixtures/sample.py
from fastapi import APIRouter
from .util import normalize

router = APIRouter()

class ChatService:
    def stream_response(self, req):
        data = normalize(req)
        return self.finish(data)

    def finish(self, data):
        return data

@router.post("/chat/stream")
def chat_stream(req):
    svc = ChatService()
    return svc.stream_response(req)
